src/pages/index.js

```diff
@@ -127,20 +182,77 @@ function handleProfileEditClick() {
 }
 
 function handleEditProfileSubmit(inputValues) {
-  userInfo.setUserInfo(profileTitleInput.value, profileDescriptionInput.value);
-  profileEditPopup.close();
+  profileEditPopup.showLoading();
+  api
+    .editProfileInfo(inputValues.title, inputValues.description)
+    .then(() => {
+      userInfo.setUserInfo(inputValues.title, inputValues.description);
+      profileEditPopup.hideLoading();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditPopup.close();
+    });
 }
 
 function handleAddCardSubmit(inputValues) {
-  const newCardData = {
-    name: addCardTitleInput.value,
-    link: addCardImageLinkInput.value,
-  };
-  const newCard = createCard(newCardData);
-  section.addItem(newCard);
-  addCardPopup.close();
+  addCardPopup.showLoading();
+  api
+    .createNewCard(inputValues.title, inputValues.link)
+    .then((res) => {
+      const newCard = createCard(res);
+      cardListSection.addItem(newCard);
+    })
+    .then((res) => {
+      addCardPopup.hideLoading();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addCardPopup.close();
+    });
 }
 
 function handleCardClick(cardData) {
   previewImagePopup.open(cardData);
 }
+
+function handleDeleteBinClick(cardID) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .deleteCard(cardID)
+      .then((res) => {
+        this.handleCardDelete();
+      })
+      .catch(console.error)
+      .finally(() => {
+        deleteCardPopup.close();
+      });
+  });
+}
+
+function handleConfirmButtonSubmit() {
+  handleCardDelete();
+}
+
+function handleCardLike(card) {
+  if (card.cardIsLiked()) {
+    api
+      .deleteCardLike(card._cardID)
+      .then((res) => {
+        card.updateLikeCount(res);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .addCardLike(card._cardID)
+      .then((res) => {
+        card.updateLikeCount(res);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors